### ISSUE TITLE
Close() doesn't appear to free resources properly.

### ISSUE THREAD

**Issue by mjgarton** [nats-io/go-nats#368]
I noticed a memory issue with an application that periodically (every few mins) opens a nats connection, does some work and closes it again.

I reproduced the issue with the below code.

```
package main

import (
	"net/http"
	_ "net/http/pprof"
	"time"

	"github.com/nats-io/go-nats"
)

func main() {
	go func() {
		panic(http.ListenAndServe(":6060", nil))
	}()

	for {
		err := openAndClose()
		if err != nil {
			panic(err)
		}
		// Don't hurt things too quickly.
		time.Sleep(10 * time.Millisecond)
	}
}

func openAndClose() error {
	conn, err := nats.Connect("http://localhost:4222", nats.Name("test-client-id-2"))
	if err != nil {
		return err
	}
	conn.Close()
	return nil
}
```

pprof heap output suggests the memory usage is due to allocations in `github.com/nats-io/go-nats.(*Conn).createConn`  which suggests (to me at least) that something is retaining a reference to the connections and the associated resources are not being freed.

**Comment by derekcollison** [comment]
Thanks will take a look.

**Comment by kozlovic** [comment]
I think I found what's wrong, will update later today.

**Comment by kozlovic** [comment]
@mjgarton Thanks for the report. I believe that we have addressed this issue in PR #370.
